xarray/testing/strategies.py

```diff
@@ -361,6 +393,23 @@ def variables(
     return xr.Variable(dims=dim_names, data=_data, attrs=draw(attrs))
 
 
+@st.composite
+def index_variables(
+    draw: st.DrawFn,
+    *,
+    dims: Union[
+        st.SearchStrategy[Union[Sequence[Hashable], Mapping[Hashable, int]]],
+        None,
+    ] = None,
+    dtype: st.SearchStrategy[np.dtype] = pandas_index_dtypes(),
+    attrs: st.SearchStrategy[Mapping] = attrs(),
+) -> xr.Variable:
+
+    index = draw(pdst.indexes(min_size=1, dtype=draw(dtype)))
+    _dims = draw(dimension_names(min_dims=1, max_dims=1))
+    return xr.Variable(dims=_dims, data=index, attrs=draw(attrs))
```

Comment:
This could be a function which returns a `st.builds(xr.Variable, ...)` rather than a composite, which is a _slight_ performance improvement but mostly just an idiom thing.